sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedFlux.java

```diff
@@ -113,8 +113,8 @@ private PagedFlux(Supplier<PageRetriever<String, PagedResponse<T>>> provider, bo
     public <S> PagedFlux<S> mapPage(Function<T, S> mapper) {
         Supplier<PageRetriever<String, PagedResponse<S>>> provider = () -> (continuationToken, pageSize) -> {
             Flux<PagedResponse<T>> flux = (continuationToken == null)
-                ? byPage()
-                : byPage(continuationToken);
+                ? byPage().take(1)
```

Comment:
I am not sure if `take(1)` is the right fix here as it would limit the flux to a single page if a mapper is applied. The right way to do it might probably be to use backpressure in PagedIterable to lazily retrieve pages.

Reply:
Thanks for reviewing.

The place calling it is at https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedByIteratorBase.java#L74-L101

From that code, it appears the `requestPage` would like to get a single page (which works as expected if provided with listSinglePage and listNextPage as Mono).
With `blockingLast` and synchronize interface, it seems hard to let `PageRetriever` to get all pages, but let that `requestPage` to take one page.

Let me know if you have good idea. I won't merge this until get consensus. Or better you guys can come up with a better fix. :-)

Reply:
@srnagar Any better idea?